_posts/2019-08-23-lti-and-saml-the-basics.markdown

```diff
@@ -0,0 +1,97 @@
+# LTI and SAML: The Basics
+
+by Cameron Gera (@gera-cameron)
+
+LTI and SAML single sign-on strategies are incredibly useful, but they are some of the hardest things to grok so I am going to put it in terms that normal developers and system administrators understand.
+These two standards have full-fledged specs out there on the internet and if you work well with those standardized papers I would encourage you to check those out.
+Sadly, every time I read those I left more confused and lost, so I am going to take a different route.
+What was most useful for me were diagrams. As I get into the details about each I will provide diagrams to explain what I am saying. So if you are looking for a simplified version of what these two standards are, then stick around.
+
+## LTI Standard
+
+The LTI standard is very common in the academic space and somewhat in the business space.
+LTI standard stands for Learning Tools Interoperability.
+Knowing this name isn't super useful besides letting you know that a learning management system (LMS) tends to integrate via this strategy.
+Tools like Canvas, Moodle, and Blackboard are examples of LMS.
+
+### How it works
+
+LTI is an SSO strategy that is always initiated from the users LMS. 
+It is a fairly streamlined process and I'll do my best to keep it clear. 
+Let's check out this diagram to visualize the flow.
+
+![LTI Diagram][]
+
+Okay, so we have two entities that will be communicating with each other.
+One is the LTI Consumer and the LTI Tool Provider.
+The LTI Consumer is the LMS.
+The LTI Tool Provider is the external service that the LMS is trying to access.
+The authentication works through a key and secret system with some other configurable pieces on the LMS side.
+
+The terms to know in regards to authentication are the following:
+
+- **Consumer Key** - This is a shared key between the Consumer and the Tool Provider.
+- **Shared Secret** - This is the privately shared secret between the Consumer and the Tool Provider.
```

Comment:
Is there a clear difference between the consumer key and the shared secret? Based on the wording I think the key might be public and the secret is private, but I'm not certain.